### PR TITLE
refactor: #158 fix eslint rule "no-unused-vars"

### PR DESCRIPTION
**Pull request by miquelbeltran**
### Description :memo:
- **Purpose**: Fixes the "no-unused-vars" lint rule.
- **Approach**: Configured the rule and performed code changes.

To allow unused parameters, this rule sets the property `args` to `none`, that way the linter will not report errors for this case, e.g. here the `req` parameter would throw an error without this configuration:

```js
raygunClient.user = function(req) {
    return 'user@example.com';
};
```

- Depends on #160 

**Type of change**

- [x] Bug fix (non-breaking change which fixes an issue)

**Updates**

- Continuing work from pr #160 
- Configure rule as explained.
- Removed several unused variables.

### Test plan :test_tube:

- run ci scripts (test and lint)
- test example apps

### Author to check :eyeglasses:
- [ ] Project and all contained modules builds successfully
- [ ] Self-/dev-tested 
- [ ] Unit/UI/Automation/Integration tests provided where applicable
- [ ] Code is written to standards
- [ ] Appropriate documentation written (code comments, internal docs)

### Reviewer to check :heavy_check_mark:
- [ ] Project and all contained modules builds successfully
- [ ] Change has been dev-/reviewer-tested, where possible
- [ ] Unit/UI/Automation/Integration tests provided where applicable
- [ ] Code is written to standards
- [ ] Appropriate documentation written (code comments, internal docs)

### CODE REVIEW

File: eslint.config.mjs [hunk 1]
```diff
@@ -8,8 +8,15 @@ export default tseslint.config(
     ...tseslint.configs.recommended,
     {
         rules: {
+            // Unused vars reported as error
+            "@typescript-eslint/no-unused-vars": [
+                "error",
+                {
+                    // Do not report unused function arguments
+                    "args": "none",
```
Comment (sumitramanga) [inline eslint.config.mjs:16]: I was wondering why we wouldn't enforce the rule to error out when unused arguments are declared. We can force the contributor to remove tech debt by having this rule on and minimize the time for this when reviewing PRs.  I would be keen to hear your thoughts :)

**Review by sumitramanga — CHANGES_REQUESTED** [review]
The code changes itself look good! Just one thing I'd like clarified please

File: eslint.config.mjs (hunk 1 above)
Comment (miquelbeltran) [reply eslint.config.mjs:16]: The problem I have found is that the error will raise as well in a situation like these:

```js
 expressCustomData(error: Error, request: Request) {
    return {};
  }
```

Here, we define a function that the user can override, but the default implementation doesn't use its parameters, and so `eslint` would complain about that.

In the README, you can see an example use of this, where the two parameters are defined:

```js
raygunClient.expressCustomData = function (err, req) {
  return { 'level': err.level };
};
```

Alternatively, we can add this rule and then add the "ignore" comment on each one of the individual cases where it should be.

Comment (sumitramanga) [reply eslint.config.mjs:16]: I see now. So this is in regard to the function needed to match the signature specified so we can't just use `expressCustomData ` without the parameters. To my understanding.

I'm happy with this justification :) We will just need to keep an eye on code reviews to ensure we aren't leaving being unused parameters that aren't needed

**Review by sumitramanga — APPROVED** [review]
LGTM 🍾

File: eslint.config.mjs (hunk 1 above)
Comment (miquelbeltran) [reply eslint.config.mjs:16]: Yes, if we see that this becomes a problem, I'll look for an alternative.

**Pull request merged**